Source/JavaScriptCore/runtime/StringPrototype.cpp

```diff
@@ -722,7 +722,7 @@ JSC_DEFINE_HOST_FUNCTION(stringProtoFuncSplitFast, (JSGlobalObject* globalObject
             for (unsigned i = 0; i < resultSize; ++i) {
                 unsigned end = result[i];
                 JSString* string = nullptr;
-                if (makeAtomStringsArray) {
+                if (makeAtomStringsArray && !isASCIIDigit(view->characterAt(start))) {
```

Comment:
When doing this, we are relying on the fact that JSImmutableButterfly with a specific Structure always have atomized string. Thus I think we need some structure change when we hit this case.